lib/publiccloud/provider.pm

```diff
@@ -16,12 +16,18 @@ use testapi;
 use Mojo::Base -base;
 use publiccloud::instance;
 use Data::Dumper;
+use Mojo::JSON 'decode_json';
 
 has key_id     => undef;
 has key_secret => undef;
 has region     => undef;
+has username   => undef;
 has prefix     => 'openqa';
 
+our $terraform_timeout = 60 * 10;
+our $terraform_dir     = '/root/terraform';
+our $terraform_applied = 0;
```

Comment:
Using `our` will not produce a member variable which we will need here.
```suggestion
has $terraform_applied => 0;
```

Reply:
done